test/Constraints/disfavored.swift

```diff
@@ -0,0 +1,37 @@
+// RUN: %target-typecheck-verify-swift
+
+struct A { }
+struct B { }
+
+@_disfavoredOverload
+func f0<T>(_: T) -> A { return A() }
+
+func f0(_: Int32) -> B { return B() }
```

Comment:
It seems like all of these are related to non-default literals, maybe we should just remove that score kind and move logic to prefer default over non-default to `CSRanking` that would enable us to prefer non-generic versions in all of examples...

Reply:
Yes, I think that's the right long-term direction.